### PR TITLE
Different documentation updates in `matrix_free`

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: include/deal.II/matrix_free/matrix_free.h [hunk 1]
```diff
@@ -95,7 +95,7 @@ DEAL_II_NAMESPACE_OPEN
  * parallelization in shared memory and vectorization.
  *
  * Vectorization is implemented by merging several topological cells into one
- * so-called macro cell. This enables the application of all cell-related
+ * so-called `cell batch'. This enables the application of all cell-related
```
Comment (kronbichler) [inline include/deal.II/matrix_free/matrix_free.h:98]:
```suggestion
 * so-called `cell batch`. This enables the application of all cell-related
```

File: include/deal.II/matrix_free/matrix_free.h [hunk 2]
```diff
@@ -130,49 +130,55 @@ class MatrixFree : public Subscriptor
...
+   * task parallel scheduling. The parameters @p mapping_update_flags,
+   * @p mapping_update_flags_boundary_faces, @p mapping_update_flags_inner_faces,
+   * and @p mapping_update_flags_faces_by_cells specify the update flags that
+   * should be stored by this class.
    *
-   * The fifth parameter specifies the level in the triangulation from which
+   * The parameter @p mg_level specifies the level in the triangulation from which
    * the indices are to be used. If the level is set to
-   * numbers::invalid_unsigned_int, the active cells are traversed, and
+   * `numbers::invalid_unsigned_int', the active cells are traversed, and
```
Comment (kronbichler) [inline include/deal.II/matrix_free/matrix_free.h:145]:
```suggestion
   * `numbers::invalid_unsigned_int`, the active cells are traversed, and
```

**Review by kronbichler — APPROVED** [review]

File: include/deal.II/matrix_free/matrix_free.h [hunk 3]
```diff
@@ -130,49 +130,55 @@ class MatrixFree : public Subscriptor
...
    * the user to disable some of the initialization processes. For example, if
    * only the scheduling that avoids touching the same vector/matrix indices
    * simultaneously is to be found, the mapping needs not be
    * initialized. Likewise, if the mapping has changed from one iteration to
    * the next but the topology has not (like when using a deforming mesh with
    * MappingQEulerian), it suffices to initialize the mapping only.
    *
-   * The two parameters `cell_vectorization_categories` and
-   * `cell_vectorization_categories_strict` control the formation of batches
+   * The two parameters @p cell_vectorization_categories and
```
Comment (kronbichler) [inline include/deal.II/matrix_free/matrix_free.h:161]: I am not sure why these changes are necessary as they do not change the appearance at all (and we might now be changing back into a state we had before), but I am not opposed to it.

Comment (peterrum) [reply include/deal.II/matrix_free/matrix_free.h:161]: I made these changes, since most of the parameters in the description used `@p`. Else I wouldn't have changed it.
